### DISCUSSION TITLE
Why does useSWRInfinite make 2 calls to the backend for every request?

### DISCUSSION THREAD

**Discussion by grantspilsbury** [vercel/swr#1612] [Q&A]
I have a load more button, within a brand new nextjs app, that uses useSWRInfinite. Every time I click it it makes 2 calls to my backend. The first call is with index 0 and the second is with the incremented index.

Here is my component:
```
function Home() {
  const { data, size, setSize } = useSWRInfinite(
    (index) => `/api/user?page=${index}&per_page=2`,
    (url) => fetch(url).then((r) => r.json()),
    { initialSize: 1 }
  )
  return <button onClick={() => setSize(size + 1)}>{'load more'}</button>
}
```
And here is my endpoint
```
export default (req, res) => {
  console.log('calling the backend with', req.query)
  res.send(200)
}
```
These are the logs after I've clicked the button 3 times:
```
calling the backend with { page: '0', per_page: '2' }
calling the backend with { page: '1', per_page: '2' }
calling the backend with { page: '0', per_page: '2' }
calling the backend with { page: '2', per_page: '2' }
calling the backend with { page: '0', per_page: '2' }
calling the backend with { page: '3', per_page: '2' }
```

Why does it call the backend with page / index 0 and then again with the correct incremented index?

**Comment by gfohl** [comment]
I have the same issue

**Comment by VewMet** [comment]
Mate, having the same issue. Its likely an swr bug. Made an infinite scroller and console logged pageIndex, and url from the fetcher function. If you see attached screenshots below (in order), you can easily spot the pattern. Needs to be reported.
![image](https://user-images.githubusercontent.com/88471847/141886511-75a6d96d-4a2e-4cab-a170-1dace2367f62.png)
![image](https://user-images.githubusercontent.com/88471847/141886574-f0d35ae8-a727-4119-81df-5ff7e07c50b8.png)
![image](https://user-images.githubusercontent.com/88471847/141886624-f056e54a-ad2f-4534-8692-1c25b5cbbc3b.png)
![image](https://user-images.githubusercontent.com/88471847/141886663-df5886a2-d546-46e2-bba4-924ff2ef7c3a.png)
![image](https://user-images.githubusercontent.com/88471847/141886739-15f18512-5ddd-4d56-9b04-bd05e6101694.png)
![image](https://user-images.githubusercontent.com/88471847/141886778-c76487f8-fd5f-4f93-bbdd-533af0ddea7c.png)

**Comment by VewMet** [comment]
raised the issue:

https://github.com/vercel/swr/issues/1638

**Comment by terminalqo** [comment]
https://github.com/vercel/swr/issues/1552

**Comment by shuding** [comment]
The behavior is explained in #1401, there is also a new option `revalidateFirstPage` in 1.1.0 that controls it.

**Reply by maria-noaptes** [reply]
Hello! I also have this problem, I don't want to refetch the first page for every new page because it implies too big of a loading time for new data in my website.
For useInfiniteSwr I also have a refreshInterval of 10 seconds, but once I put revalidateFirstPage on false it doesn't update anymore in an interval.
Is there any way to have live updates but not revalidation for the first page at load more?
